R/QFeatures-constructors.R

```diff
@@ -1,85 +1,233 @@
+##' @seealso
+##'
+##' - The `QFeatures` (see [QFeatures()]) class to read about how to
+##'   manipulate the resulting `QFeatures` object.
+##'
+##' - The [readQFeaturesFromDIANN()] function to import DIA-NN
+##'   quantitative data.
 ##'
 ##' @md
+##'
+##' @name readQFeatures
 ##' @aliases readSummarizedExperiment
+##' @aliases readQFeatures
+##' @aliases readQFeatures,data.frame,data.frame
+##' @aliases readQFeatures,data.frame,vector
+##' @aliases readQFeatures,missing,vector
 ##' @export
 ##'
 ##' @examples
 ##'
+##' ######################################
+##' ## Single-set case.
+##'
 ##' ## Load a data.frame with PSM-level data
 ##' data(hlpsms)
+##' hlpsms[1:10, c(1, 2, 10:11, 14, 17)]
 ##'
 ##' ## Create the QFeatures object
-##' fts2 <- readQFeatures(hlpsms, ecol = 1:10, name = "psms")
-##' fts2
-readQFeatures <- function(table, ecol, fnames, ..., name = NULL)  {
-    se <- readSummarizedExperiment(table, ecol, fnames, ...)
-    if (anyDuplicated(rownames(se))) {
-        message("Making assay rownames unique.")
-        rownames(se) <- make.unique(rownames(se))
-    }
-    cd <- DataFrame(row.names = colnames(se))
-    if (is.null(name))
-        name <- "features"
-    el <- structure(list(se), .Names = name[1])
-    al <- AssayLinks(AssayLink(name = name[1]))
-    ans <- MultiAssayExperiment(el, colData = cd)
-    new("QFeatures",
-        ExperimentList = ans@ExperimentList,
-        colData = ans@colData,
-        sampleMap = ans@sampleMap,
-        metadata = ans@metadata,
-        assayLinks = al)
-}
-##' @describeIn readQFeatures Convert tabular data from a spreadsheet or a
-##' `data.frame` into a `SummarizedExperiment` object.
+##' qf1 <- readQFeatures(hlpsms, colAnnotation = 1:10, name = "psms")
+##' qf1
+##' colData(qf1)
+##'
+##' ######################################
+##' ## Single-set case using a data.frame
+##'
+##' ## All PSMs were acquired in the same acquisition
+##' hlpsms$file <- "File1"
+##' hlpsms[1:10, c(1, 2, 10:11, 14, 17, 29)]
+##' (colann <- data.frame(file = rep("File1", 10),
+##'                       Channel = names(hlpsms)[1:10]))
+##'
+##' qf2 <- readQFeatures(hlpsms, colAnnotation = colann,
+##'                      batchCol = "file", channelCol = "Channel")
+##' qf2
+##' colData(qf2)
+##'
+##' ######################################
+##' ## Multi-set case.
+##'
+##' ## Let's simulate 3 different files/batches for that same input
+##' ## data.frame, and define a colAnnotation data.frame.
+##'
+##' hlpsms$file <- paste0("File", sample(1:3, nrow(hlpsms), replace = TRUE))
+##' hlpsms[1:10, c(1, 2, 10:11, 14, 17, 29)]
+##' (colann <- data.frame(file = rep(paste0("File", 1:3), each = 10),
+##'                       Channel = rep(names(hlpsms)[1:10], 3)))
+##'
+##' qf3 <- readQFeatures(hlpsms, colAnnotation = colann,
+##'                      batchCol = "file", channelCol = "Channel")
+##' qf3
+##' colData(qf3)
+NULL
+
+## Simple case, with a single table to populate one QFeatures set
```

Comment:
What's the purpose of this line?

Reply:
Comments to relate the code to the docs. I have udpated it to clarify.